s3/s3_operations.cpp

```diff
@@ -1144,6 +1146,56 @@ namespace irods_s3 {
             return ret;
         }
 
+        // issue 1978 - If they called open/create without close, the object will not yet exist in s3.
+        if (file_open_create_without_close) {
+
+            // The object does not yet exist on S3 or the object that is in S3 is not up-to-date because close has
+            // not been called.  If we are using a cache file, do a stat() of that.  Otherwise, return the maximum
+            // byte written.
+
+            namespace transport = irods::experimental::io::s3_transport;
+            namespace bf = boost::filesystem;
+
+            _statbuf->st_mode = S_IFREG;
+            _statbuf->st_nlink = 1;
+            _statbuf->st_uid = getuid ();
+            _statbuf->st_gid = getgid ();
+            _statbuf->st_atime = _statbuf->st_mtime = _statbuf->st_ctime = std::chrono::system_clock::now().time_since_epoch().count();
+
+            // attempt to get size from cache file
+            std::string s3_cache_dir_str = get_cache_directory(_ctx.prop_map());
+            bf::path cache_file =  bf::path(s3_cache_dir_str) / bf::path(key + "-cache");
+            struct stat st;
+
+            if (0 == stat(cache_file.string().c_str(), &st)) {
+
+                _statbuf->st_size = st.st_size;
+
+            } else {
+
+                // Not using a cache file.  Read the max byte written from interprocess shmem.
+
+                std::string bucket_name;
+                std::string shmem_key = irods::experimental::io::s3_transport::constants::SHARED_MEMORY_KEY_PREFIX +
+                    std::to_string(std::hash<std::string>{}(resource_name + "/" + key));
+
+
+                irods::experimental::interprocess::shared_memory::named_shared_memory_object<transport::shared_data::multipart_shared_data>
```

Comment:
Consider using a type or namespace alias here for readability.

Reply:
done